R/nm-file.R

```diff
@@ -0,0 +1,136 @@
+#' Read NONMEM files
+#'
+#' Reads in a whitespace-delimited NONMEM output file (for example .grd or .ext
+#' or a table output) or a NONMEM input data file.
+#'
+#' @return A tibble with the data from the specified file and estimation method.
+#'
+#' @param .mod Either a `bbi_nonmem_model`, `bbi_nonmem_summary`, or a path to a
+#'   file to read in. If passing model object to `nm_file()`, must also pass `.suffix` that
+#'   will be passed through to [build_path_from_model()].
+#' @param .est_method If `NULL`, the default, pulls the data from the final
+#'   estimation method. If an integer, pulls the data from that estimation
+#'   method. Note that, if more than one table is written to a given file, this
+#'   will also mean it pulls the final table by default, and this argument can
+#'   be used to pull a different table instead.
+#' @inheritParams build_path_from_model
+#' @param ... arguments passed through to methods. (Currently none.)
```

Comment:
What's the advantage of adding `...` now rather than waiting to see if a need arises? I haven't been able to come up with why adding it later would be a breaking change.

Reply:
I don't think it's a big deal either way. I always do it by habit because it means, when you extend the method to other classes later, you don't have to change the signature of the original generic (other existing methods). This is obviously more important for methods that are likely to be extended in other packages, but it felt like a good practice to follow. Based on [this section](https://adv-r.hadley.nz/s3.html) of the Hadley book:

<img width="718" alt="Screen Shot 2021-11-08 at 8 57 11 AM" src="https://user-images.githubusercontent.com/7806083/140755010-140d9575-bd1f-48a4-8e15-06ce32573766.png">